### PR TITLE
chore: update mypy config for project.

### PR DESCRIPTION
**Pull request by peterschutt**
- update `additional_dependencies` per project deps
- remove pydantic plugin activation

[//]: # "By submitting this pull request, you agree to:"
[//]: # "- follow [Litestar's Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow [Litestar's contribution guidelines](https://github.com/litestar-org/.github/blob/main/CONTRIBUTING.md)"

### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description
[//]: # "Please describe your pull request for new release changelog purposes"

- 

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

**Comment by peterschutt** [comment]
I noticed that pre-commit wasn't checking the `app.py` module - however, it seems to be after these changes. I'm not sure what change here actually fixed it, but I think they make sense.

**Comment by peterschutt** [comment]
image
https://github.com/litestar-org/litestar-hello-world/actions/runs/6210130228/job/16858026306?pr=3#step:6:26

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,25 +59,8 @@ repos:
...
-            pytest-timeout,
-            pytest_docker,
-            redis,
-            rich,
-            rich-click,
-            click,
-            uvicorn,
-          ]
+        exclude: "tools|docs"
+        additional_dependencies: ["litestar[standard]", pytest]
```
Comment (JacobCoffee) [inline .pre-commit-config.yaml:63]: Will `["litestar[standard]"` include click/rich/rich-click, uvicorn/etc? idk how these mypy dep things work tbh

**Review by JacobCoffee — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (peterschutt) [reply .pre-commit-config.yaml:63]: This is just emulating the dependencies we define in pyproject for the app.

**Pull request merged**